gc/gc-tool/src/main/java/org/projectnessie/gc/tool/cli/options/MarkOptions.java

```diff
@@ -0,0 +1,138 @@
+        "- an ISO instant"
+      })
+  Map<String, String> cutoffPolicies = new LinkedHashMap<>();
+
+  @CommandLine.Option(
+      names = "--write-live-set-id-to",
+      description = "Optional, the file name to persist the created live-set-id to.")
+  Path liveSetIdFile;
+
+  @CommandLine.Option(
+      names = "--identify-parallelism",
+      defaultValue = "4",
+      description = "Number of Nessie references that can be walked in parallel.")
+  int parallelism;
+
+  @CommandLine.Spec CommandSpec commandSpec;
+
+  public NessieOptions getNessie() {
+    return nessie;
+  }
+
+  public int getParallelism() {
+    return parallelism;
+  }
+
+  public Path getLiveSetIdFile() {
+    return liveSetIdFile;
+  }
+
+  /**
+   * Parses the {@code --cutoff} parameters, syntax is {@code ref-name-regex '=' number-of-commits |
+   * duration | ISO-instant}.
+   */
+  public PerRefCutoffPolicySupplier createPerRefCutoffPolicySupplier() {
+    List<Entry<Predicate<String>, CutoffPolicy>> policies =
+        cutoffPolicies.entrySet().stream()
+            .map(
+                e -> {
+                  Predicate<String> predicate = Pattern.compile(e.getKey()).asPredicate();
+
+                  Exception ex;
+                  try {
+                    return Maps.immutableEntry(
+                        predicate, CutoffPolicy.numCommits(Integer.parseInt(e.getValue())));
+                  } catch (NumberFormatException f) {
+                    ex = f;
+                  }
+                  try {
+                    return Maps.immutableEntry(
+                        predicate,
+                        CutoffPolicy.atTimestamp(
+                            cutoffPolicyRefTime.minus(Duration.parse(e.getValue()))));
+                  } catch (DateTimeException f) {
+                    ex.addSuppressed(f);
+                  }
+                  try {
+                    return Maps.immutableEntry(
+                        predicate,
+                        CutoffPolicy.atTimestamp(
+                            Instant.from(DateTimeFormatter.ISO_INSTANT.parse(e.getValue()))));
+                  } catch (DateTimeException f) {
+                    ex.addSuppressed(f);
+                  }
+                  throw new ParameterException(
+                      commandSpec.commandLine(),
+                      "Failed to parse cutoff-value '"
+                          + e.getValue()
+                          + "' for key '"
+                          + e.getKey()
+                          + "', must be either the number of commits, a duration (as per java.time.Duration) or an ISO instant (like 2011-12-03T10:15:30Z)",
+                      ex);
+                })
+            .collect(Collectors.toList());
+
+    return namedReference ->
+        policies.stream()
+            .filter(e -> e.getKey().test(namedReference.getName()))
+            .map(Entry::getValue)
+            .findFirst()
+            .orElse(CutoffPolicy.NONE);
```

Comment:
While playing with the tool. I felt we need to have default cutoff time configurations.

For example, If I have 100 branch that needs gc, I can't pass each branch name with cutoff time to the cutoff argument.